src/engine/sidechain/engineshoutcast.cpp

```diff
@@ -455,6 +496,45 @@ bool EngineShoutcast::serverConnect() {
 
 void EngineShoutcast::write(unsigned char *header, unsigned char *body,
                             int headerLen, int bodyLen) {
+    struct shoutcastCacheObject *l_SCacheObj = (struct shoutcastCacheObject *) malloc(sizeof(struct shoutcastCacheObject));
```

Comment:
That should be no problem. But do we actually need that much? Can't we pass all data instantly to the server? In which case the buffer is required?